shell/platform/tizen/tizen_event_loop.cc

```diff
@@ -66,39 +69,28 @@ void TizenEventLoop::PostTask(FlutterTask flutter_task,
   task.order = ++task_order_;
   task.fire_time = TimePointFromFlutterTime(flutter_target_time_nanos);
   task.task = flutter_task;
-  if (ecore_pipe_) {
-    ecore_pipe_write(ecore_pipe_, &task, sizeof(task));
+  {
+    std::lock_guard<std::mutex> lock(task_queue_mutex_);
+    task_queue_.push(task);
   }
-}
 
-Eina_Bool TizenEventLoop::TaskTimerCallback(void* data) {
-  auto* self = reinterpret_cast<TizenEventLoop*>(data);
-  self->ExcuteTaskEvents();
-  return EINA_FALSE;
-}
-
-void TizenEventLoop::ExcuteTaskEvents(void* data,
-                                      void* buffer,
-                                      unsigned int nbyte) {
-  auto* self = reinterpret_cast<TizenEventLoop*>(data);
-  auto* p_task = reinterpret_cast<Task*>(buffer);
-
-  const double flutter_duration =
-      (static_cast<double>(p_task->fire_time.time_since_epoch().count()) -
-       self->get_current_time_()) /
-      1000000000.0;
+  const auto flutter_duration =
+      static_cast<double>(flutter_target_time_nanos) - get_current_time_();
   if (flutter_duration > 0) {
-    {
-      std::lock_guard<std::mutex> lock(self->task_queue_mutex_);
-      self->task_queue_.push(*p_task);
-    }
-    ecore_timer_add(flutter_duration, TaskTimerCallback, self);
+    ecore_timer_add(
+        flutter_duration / 1000000000.0,
+        [](void* data) -> Eina_Bool {
+          auto* self = reinterpret_cast<TizenEventLoop*>(data);
+          if (self->ecore_pipe_) {
+            ecore_pipe_write(self->ecore_pipe_, nullptr, 0);
+          }
+          return ECORE_CALLBACK_CANCEL;
+        },
+        this);
   } else {
-    {
-      std::lock_guard<std::mutex> lock(self->expired_tasks_mutex_);
-      self->expired_tasks_.push_back(p_task->task);
+    if (ecore_pipe_) {
+      ecore_pipe_write(ecore_pipe_, nullptr, 0);
```

Comment:
Is there a better way to "notify" the ecore thread without having to write `nullptr` to the pipe? There's no data to send to the thread.

Reply:
No matter what we do, it probably won't be as simple and intuitive as this one.
I also wonder if there is another way. :eyes: